src/main/java/GenericFunction.java

```diff
@@ -1,26 +1,33 @@
 import java.io.*;
+import java.util.List;
```

Comment:
in this file there's a place where you swallow all exceptions.  this seems bad.  also separate out  testing code plz